t/05-distribution.t

```diff
@@ -61,6 +61,35 @@ subtest 'script_run' => sub {
     is $cmdcall->{buffer_size}, 6272, 'appropriate buffer size used for long command';
 };
 
+subtest 'set expected serial and autoinst failures' => sub {
+    my $d = distribution->new;
+    # Define the expected failures data
+    my @failures = (
+        {type => 'soft', message => 'Soft %s Failure Message 1', pattern => 'Test Pattern1'},
+        {type => 'soft', message => 'Soft %s Failure Message 2', pattern => 'Test Pattern2'},
+        {type => 'hard', message => 'Hard %s Failure Message 3', pattern => 'Test Pattern3'},
+    );
+    # Subroutine to generate failure data with formatted messages
+    my sub _generate_failures {
+        my $type = shift;
+        return [
+            map {
+                {
+                    %{$_},
+                      message => sprintf($_->{message}, $type),
+                      pattern => qr/$_->{pattern}/
+                }
+            } @failures
+        ];
+    }
```

Comment:
```suggestion
    my sub _generate_failures ($type, %details) {[map {{
                      %details,
                      message => sprintf($details{message}, $type),
                      pattern => qr/$details{pattern}/
                }
            } @failures
        ]}
```

Reply:
I will look into it, and apply the suggestion shortly.